src/widgets/table.rs

```diff
@@ -602,8 +602,13 @@ impl TableState {
         &mut self.offset
     }
 
-    pub fn with_selected(mut self, selected: Option<TableSelection>) -> Self {
-        self.selected = selected;
+    pub fn with_selected(mut self, selected: Option<impl Into<TableSelection>>) -> Self {
```

Comment:
This didn't work with the new "demo2".. It is now working as expected, though if this is merged, we should take the time to change all the `usize`s for `TableSelection`s in ratatui's examples

Reply:
Done!